src/main/java/dk/kb/datahandler/oai/OaiJobCache.java

```diff
@@ -96,6 +97,15 @@ public static List<OaiJobDto> convertToDto(List<OaiTargetJob> jobs){
 
     }
 
+    public static boolean isJobRunningForTarget(String targetName) {
```

Comment:
Needs to be `synchronized` to avoid getting ConcurrentModificationExceptions. The same goes for all other methods accessing the maps.

Reply:
Still need to synchronize `getRunningJobsMostRecentFirst` and `getCompletedJobsMostRecentFirst` as they also access the maps.

Reply:
fixed